tests/ingest_runs/test_tasks.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+from unittest.mock import call
+from creator.files.models import File, FileType
+from creator.ingest_runs.models import IngestRun
+from creator.ingest_runs.tasks import (
+    run_ingest,
+    cancel_ingest,
+    cancel_duplicate_ingest_runs,
+    get_ingest_run,
+)
+
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+@pytest.fixture
+def mock_enqueue(mocker):
+    """
+    Mock the django_eq.enqueue function.
+    """
+    enqueue = mocker.patch("creator.ingest_runs.tasks.django_rq.enqueue")
+    return enqueue
+
+
+def test_run_ingest(db, mocker, clients, prep_file, mock_enqueue):
+    """
+    Test the run_ingest function.
+    """
+    client = clients.get("Administrator")
+    mock_genomic_workflow = mocker.patch(
+        "creator.ingest_runs.tasks.ingest_genomic_workflow_output_manifests"
+    )
+    user = User.objects.first()
+
+    # Create data. The last Version will have a non-GWO FileType
+    for _ in range(3):
+        prep_file(authed=True)
+    files = list(File.objects.all())
+    for file_ in files[:-1]:
+        file_.file_type = FileType.GWO.value
+        file_.save()
+    file_versions = [f.versions.first() for f in files]
+
+    """
+    1) Happy Case
+    Call run_ingest. Check that genomic_file_workflow got
+    called. Check that ir.state == 'completed'. Call run_ingest again
+    and make sure that _django_rq.enqueue(cancel_ingest, ingest_id)_ is called.
+    """
+    happy_versions = file_versions[:-1]
+    # An initial IngestRun with no issues
+    happy_ir = setup_ingest_run(happy_versions, user)
+    run_ingest(happy_ir.id)
+
+    assert count_ingest_runs() == 1
+    mock_genomic_workflow.assert_called_once()
+    mock_genomic_workflow.reset_mock()
+    assert get_ingest_run(happy_ir.id).state == "completed"
+
+    # Call run_ingest on a duplicate IngestRun. The older IngestRun should
+    # be canceled, and the newer one gets deleted.
+    old_ir = setup_ingest_run(happy_versions, user)
+    old_ir.start()
+    old_ir.save()
+    assert count_ingest_runs() == 2
+
+    new_ir = setup_ingest_run(happy_versions, user)
+    assert count_ingest_runs() == 3
+    run_ingest(new_ir.id)
+    # The above call should delete _new_ir_
+    assert count_ingest_runs() == 2
+
+    mock_enqueue.assert_called_once_with(cancel_ingest, old_ir.id)
+    mock_enqueue.reset_mock()
+    assert new_ir.state == "waiting"
+
+    with pytest.raises(Exception):
+        assert get_ingest_run(new_ir.id)
+
+    """
+    2) Non-GWO Case
+    Call run_ingest on an IngestRun with a version that doesn't have a GWO
+    root_file. Exception should be raised.
+    """
+    bad_ir = setup_ingest_run(file_versions, user)
+    with pytest.raises(Exception):
+        assert run_ingest(bad_ir.id)
+
```

Comment:
Once you add the try/except around the call to `ingest_genomic_worflow_manifest`, we should probably have a test that mocks that function and has an exception side effect so that we can check that the ingest_run goes to fail state